### PR TITLE
Make it possible to go from CMS to page

### PR DESCRIPTION
**Pull request by Mackiovello**
Added a button on the `/content/cms` page that makes it possible to go to the specified page.  For #27 

Faced a weird error with `puppet-redirect` where using the `history` attribute caused the body margin to not be applied when clicking the button but only when manually navigating there.

### CODE REVIEW

File: src/Content/wwwroot/Content/viewmodels/CmsPage.html [hunk 1]
```diff
@@ -77,13 +77,14 @@
...
                                 <button type="button" class="btn btn-xs btn-default" item="{{item}}" on-click="editItem">Edit</button>
                                 <button type="button" class="btn btn-xs btn-default" value="{{item.Delete$::click}}" onmousedown="++this.value;">Delete</button>
+                                <button type="button" class="btn btn-xs btn-default" item="{{item}}" on-click="goToItem">Go to item</button>
                             </td>
                         </tr>
                     </template>
                 </tbody>
             </table>
         </fieldset>
-        <link is="puppet-redirect" history url$="{{model.RedirectUrl$}}" />
```
Comment (warpech) [inline src/Content/wwwroot/Content/viewmodels/CmsPage.html]: Removing `history` attribute is a code smell. It forces full page reload, which should not be needed here. 

Can you pls restore the attribute and create a new issue for the problem that it causes? Please cc @tomalec

File: src/Content/wwwroot/Content/viewmodels/CmsPage.html [hunk 2]
```diff
@@ -77,13 +77,14 @@
                             <td class="content-cms-table__td-actions">
                                 <button type="button" class="btn btn-xs btn-default" item="{{item}}" on-click="editItem">Edit</button>
                                 <button type="button" class="btn btn-xs btn-default" value="{{item.Delete$::click}}" onmousedown="++this.value;">Delete</button>
+                                <button type="button" class="btn btn-xs btn-default" item="{{item}}" on-click="goToItem">Go to item</button>
```
Comment (warpech) [inline src/Content/wwwroot/Content/viewmodels/CmsPage.html]: Maybe call the button "Preview"?

**Comment by warpech** [comment]
> Faced a weird error with `puppet-redirect` where using the `history` attribute caused the body margin to not be applied when clicking the button but only when manually navigating there.

Please always try to investigate and fix or report *weird errors* at the root, not by bypassing them. 

Suggested approach: isolate the problem, report it and try to get it fixed fast. If a short term solution is absolutely necessary, then apply a workaround, but always make sure that we are tracking the root problem, too.

Thanks!

**Comment by Mackiovello** [comment]
This is confusing. I added the `history` attribute and I'm not facing the issue anymore. The issue was that the `body { margin: 20px };` from the middleware was not applied when reaching the page by clicking the button and not when entering the URI. It's possible that it was a local problem that was solved by rebooting + reinstalling Starcounter.

**Comment by Mackiovello** [comment]
I've created an issue for it https://github.com/Palindrom/palindrom-redirect/issues/6

The history attribute is added and the button is renamed. Should I go ahead and merge or should we wait until there is a fix?

File: src/Content/wwwroot/Content/viewmodels/CmsPage.html [hunk 3]
```diff
@@ -77,14 +77,14 @@
...
-                                <button type="button" class="btn btn-xs btn-default" item="{{item}}" on-click="goToItem">Go to item</button>
+                                <button type="button" class="btn btn-xs btn-default" item="{{item}}" on-click="goToItem">View</button>
                             </td>
                         </tr>
                     </template>
                 </tbody>
             </table>
         </fieldset>
-        <link is="puppet-redirect" url$="{{model.RedirectUrl$}}" />
+        <link is="puppet-redirect" hisory url$="{{model.RedirectUrl$}}" />
```
Comment (warpech) [inline src/Content/wwwroot/Content/viewmodels/CmsPage.html]: history

**Pull request merged**